### ISSUE TITLE
Pass -fno-it on GHC 8.6

### ISSUE THREAD

**Issue by ndmitchell** [ndmitchell/ghcid#130]
See https://ghc.haskell.org/trac/ghc/ticket/14336, which should allow for smaller memory leaks.

**Comment by ndmitchell** [comment]
Note that this requires dynamically detecting the GHC version. which might be tricky... Looking for the initial `GHC version ...` line is probably easiest.

**Comment by ndmitchell** [comment]
Not harmful on earlier GHC's to pass and forget.

**Comment by ndmitchell** [comment]
I undid the fix due to GHC bugs

**Comment by fosskers** [comment]
Still relevant?

**Comment by ndmitchell** [comment]
Yes. GHC introduced `-no-it` which reduces memory usage in Ghci, which would be great. It's available from GHC 8.6 and above. Unfortunately, this flag also turns on the behaviour of printing out lots of spurious `()` values. Ghcid isn't expecting those, and gets upset. Because GHC 8.6 has those spurious `()` values the options are 1) never use `-no-it`, 2) cope with `()`, 3) patch GHC to not print out the spurious `()` and wait a decade until Ghcid no longer supports the last version which prints them. Of those, only 2 seems feasible, but I haven't done the work for it.
